Us20 implement manageBadgesEndpointTest

AmineChalouati opened the pull request.

AmineChalouati committed:
US 5  Implement UNit tests

AmineChalouati committed:
US 5: GamificationService Unit Test
* Add test use cases
* Remove unused dependency

AmineChalouati committed:
US 5:
*Reformat code
*Optimize import

AmineChalouati committed:
US 20:ImplementManageBadgesENdPoint  webservices Test

bourasom commented on services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java:
```diff
@@ -0,0 +1,153 @@
+        org.junit.Assert.assertTrue(new ArrayList().isEmpty());
+        LOG.info("****Initiation of the assert of badges List**** ", BadgeEntity.class);
+
+        HttpUriRequest request = new HttpGet("http://localhost:8080/rest/gamification/badges/all");
```
Ends with org.apache.http.conn.HttpHostConnectException connection refused. Are you sure that something is runing under localhost:8080 when tests are executed? Should foccus how to mock the request

AmineChalouati replied:
PR updated

bourasom commented on services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java:
```diff
@@ -0,0 +1,153 @@
+        LOG.info("****Initiation of the assert of badges List**** ", BadgeEntity.class);
+
+        HttpUriRequest request = new HttpGet("http://localhost:8080/rest/gamification/badges/all");
+        assertNotEquals(request, new ArrayList().isEmpty());
```
What is the relation between an empty ArrayList and request?

AmineChalouati replied:
PR updated

bourasom commented on services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java:
```diff
@@ -0,0 +1,153 @@
+
+        // When
+        HttpResponse httpResponse = HttpClientBuilder.create().build().execute(request);
+        assertThat(
```
Not reashed. Exceptions are rised before

bourasom commented on services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java:
```diff
@@ -0,0 +1,153 @@
+
+        LOG.info("List of badges is OK ", BadgeEntity.class, httpResponse.getStatusLine());
+
+        assertNotEquals(HttpStatus.SC_NOT_FOUND, HttpStatus.SC_OK);
```
Please remove this evident assert. The two constants are differents within and without the test

bourasom commented on services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java:
```diff
@@ -0,0 +1,153 @@
+            HttpUriRequest request = new HttpGet("http://localhost:8080/rest/gamification/badges/add" + Badgename);
+
+            // When
+            HttpResponse response = HttpClientBuilder.create().build().execute(request);
```
Will rise an exception and the following parts will not be executed

bourasom commented on services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java:
```diff
@@ -0,0 +1,153 @@
+
+            // Then
+            String mimeType = ContentType.getOrDefault(response.getEntity()).getMimeType();
+            assertEquals(jsonMimeType, jsonMimeType);
```
Always true

bourasom commented on services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java:
```diff
@@ -0,0 +1,153 @@
+            HttpUriRequest request = new HttpGet("http://localhost:8080/rest/gamification/badges/delete" + badgeTitle);
+
+            // When
+            HttpResponse response = HttpClientBuilder.create().build().execute(request);
```
Will rise an exception and the following parts will not be executed

bourasom commented on services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java:
```diff
@@ -0,0 +1,153 @@
+
+            // Then
+            String mimeType = ContentType.getOrDefault(response.getEntity()).getMimeType();
+            assertEquals(jsonMimeType, jsonMimeType);
```
Always true

bourasom commented on services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java:
```diff
@@ -0,0 +1,153 @@
+        String jsonMimeType = "application/json";
+        try {
+            // Given
+            HttpUriRequest request = new HttpGet("http://localhost:8080/rest/gamification/badges/update");
```
will rise exception

bourasom commented on services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java:
```diff
@@ -0,0 +1,153 @@
+            // Then
+            String mimeType = ContentType.getOrDefault(response.getEntity()).getMimeType();
+            assertEquals(jsonMimeType, jsonMimeType);
+            LOG.info("The update of the selected badge is completely done with success ", BadgeEntity.class, response.getStatusLine());
```
Always true

bourasom commented on services/src/test/java/org/exoplatform/addons/gamification/test/AbstractServiceTest.java:
```diff
@@ -14,29 +15,31 @@
 import org.exoplatform.social.core.space.spi.SpaceService;
```
What is the purpose of this class?

AmineChalouati committed:
US 20: implement ManageBadgesEndpoint Unit test
Implement getAllBadges()/addBadge()/updateBadge()/deleteBadge()
Methods test
Add the required configuration of the used components

AmineChalouati committed:
Add getcomponent class to TestGamificationService

Add getcomponent class to US5 unit test Gamification in TestGamificationService

AmineChalouati committed:
Update TestGamificationService.java

AmineChalouati committed:
US 20: implement ManageBadgesEndpoint Unit test
add Gamification Mock file + configuration or the server port and name

AmineChalouati committed:
Merge remote-tracking branch 'origin/US20ImplementMangeBadgesENdPoint' into US20ImplementMangeBadgesENdPoint

# Conflicts:
#	services/src/test/java/org/exoplatform/addons/gamification/service/TestGamificationService.java

AmineChalouati committed:
correct line in gamification Test service

AmineChalouati committed:
Update the ManageBadgesEndpoint Unit test + add the changement in the Manage Badge point web service because of a not nullable Iconfield that can block the add of a new badge

AmineChalouati commented on services/src/main/java/org/exoplatform/addons/gamification/rest/ManageBadgesEndpoint.java:
```diff
@@ -15,283 +27,253 @@
-                badgeDTO.setIconFileId(fileItem.getFileInfo().getId());
+          }
+          /** END upload */
+          badgeDTO.setIconFileId(fileItem.getFileInfo().getId());
```
@bourasom this is the changement in the webservice

AmineChalouati committed:
US 20: Last new changement in names of classes

AmineChalouati committed:
change Connot to Cannot

bourasom reviewed (commented):
LGTM

AmineChalouati committed:
Merge branch 'master' into US20ImplementMangeBadgesENdPoint
